LLama/Model/ModelCache.cs

```diff
@@ -10,25 +10,35 @@
 
 namespace LLama.Model;
 
+internal class CachedModelReference
```

Comment:
Since the underlying `SafeHandle` already has a refcount is this needed? It looks like it's only used in the `UnloadModel`, but I'm not quite sure what for, so I might just be missing something about the lifetime management.

Reply:
removed per discussion above